### ISSUE TITLE
question: is the creation of dynamic gridlayouts supported

### ISSUE THREAD

**Issue by DavyDeDurpel** [NativeScript/nativescript-angular#661]
I'm trying to create a GridLayout dynamically.  I've been experimenting with a custom directive and the dynamic creation of custom angular components.  But I can't seem to be able to figure out how to link NS View with NG Component.
The best I can do for the moment is setting some properties of the GridLayout dynamically and adding 4 angular components.  The problem is that the components are rendered on top of each other instead of in a nice grid.

Can someone help me?

Below is what I have so far.  
ChildComponent, DynamicGrid and myDirective should be added to the declarations.
ChildComponent should be added to the entryComponents of the NgModule

--------------------------------------------------------------------

```
import { Component, Directive, Inject, ViewContainerRef, TemplateRef, ComponentFactoryResolver, Input, ComponentRef } from "@angular/core";
import { GridLayout, ItemSpec } from 'ui/layouts/grid-layout';
import { Color } from 'color';
import { Label } from 'ui/label';

@Component({
    selector: "main",
    template: "<GridLayout *myDirective ></GridLayout>"
})
export class DynamicGrid {

    constructor() { }
}


@Directive({ selector: "[myDirective]" })
export class myDirective {
    constructor(private container: ViewContainerRef, private templateRef: TemplateRef<any>, private componentFactoryResolver: ComponentFactoryResolver) {}

    ngOnInit() {
        const viewRef = this.container.createEmbeddedView(this.templateRef);

        //filter out whitespace nodes
        const gridViews = viewRef.rootNodes.filter((node) => node && node.nodeName !== '#text');

        if (gridViews.length > 0) {
            const grid: GridLayout = gridViews[0];
            grid.borderColor = new Color("orange");
            grid.borderWidth = 1;
            grid.addColumn(new ItemSpec(1, "auto"));
            grid.addColumn(new ItemSpec(1, "auto"));
            grid.addRow(new ItemSpec(1, "auto"));
            grid.addRow(new ItemSpec(1, "auto"));

            let factory = this.componentFactoryResolver.resolveComponentFactory(ChildComponent);
            let res: ComponentRef<ChildComponent> = this.container.createComponent(factory);
            res.instance.text = "Col 1, Row 1";
            res.instance.col = "1";
            res.instance.row = "1";

            res = this.container.createComponent(factory);
            res.instance.text = "Col 2, Row 1";
            res.instance.col = "2";
            res.instance.row = "1";

            res = this.container.createComponent(factory);
            res.instance.text = "Col 1, Row 2";
            res.instance.col = "1";
            res.instance.row = "2";

            res = this.container.createComponent(factory);
            res.instance.text = "Col 2, Row 2";
            res.instance.col = "2";
            res.instance.row = "2";
        }
    }

}

@Component({
    selector: 'my-child',
    template: "<StackLayout [col]='col' [row]='row'><Label [text]='text'></Label></StackLayout>"
})
export class ChildComponent {
    @Input() text: string;
    @Input() col: string;
    @Input() row: string;
}
```

**Comment by johnnyzen** [comment]
I'm having similar issue:

Have you checked out http://www.nativescriptsnacks.com/snippets/2016/06/20/child-components-gridlayout.html?

**Comment by matt4446** [comment]
An extremely long time ago (10 months ago) I answered someone with the same sort of question. So some angular2 syntax has probably changed as well as some of the nativescript references. You should be able to compare and add to yours though (saying that alot changed between RC and final). 

https://github.com/matt4446/NS-Angular-GridTest/blob/master/app/app.component.ts#L12-L13
https://github.com/matt4446/NS-Angular-GridTest/blob/master/app/controls/grid.directive.ts#L6-L53

I plan to start updating my examples soon ;-) 
Free time is coming.

**Comment by DavyDeDurpel** [comment]
Thanks both for the examples.  I'll give it a try this week.

**Comment by johnnyzen** [comment]
In the snack link I posted, I do not understand this comment:

// Pass in your element to these static functions and they will attach to the closest parent gridlayout

I wonder if this means I cannot use a factory to inject an component into the gridlayout directly?

**Comment by DavyDeDurpel** [comment]
@johnnyzen I couldn't get this to work.  I didn't do a lot of effort to get it working though.

@matt4446 I was able to get your example working.  In fact I only needed to refactor the 'directives' part.  You no longer need to specify them on the component.  You just need to add the 2 directives to the declarations of the ngModule.

I already modified the example to my needs.  I added a ListView with template selector inside the cells of the grid.  I also added support for colspan.  I will cleanup my example and post it tomorrow.
